providers/dns/bluecat/internal/client.go

```diff
@@ -0,0 +1,286 @@
+package internal
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+)
+
+// Object types.
+const (
+	ConfigType = "Configuration"
+	ViewType   = "View"
+	ZoneType   = "Zone"
+	TXTType    = "TXTRecord"
+)
+
+type Client struct {
+	HTTPClient *http.Client
+
+	baseURL string
+	token   string
+}
+
+func NewClient(baseURL string) *Client {
+	return &Client{
+		baseURL:    baseURL,
+		HTTPClient: &http.Client{Timeout: 30 * time.Second},
+	}
+}
+
+// Login Logs in as API user.
+// Authenticates and receives a token to be used in for subsequent requests.
+// https://docs.bluecatnetworks.com/r/Address-Manager-API-Guide/GET/v1/login/9.1.0
+func (c *Client) Login(username, password string) error {
+	queryArgs := map[string]string{
+		"username": username,
+		"password": password,
+	}
+
+	resp, err := c.sendRequest(http.MethodGet, "login", nil, queryArgs)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		data, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("status code: %d, message: %s", resp.StatusCode, string(data))
+	}
+
+	authBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	authResp := string(authBytes)
+	if strings.Contains(authResp, "Authentication Error") {
+		return fmt.Errorf("request failed: %s", strings.Trim(authResp, "\""))
+	}
+
+	// Upon success, API responds with "Session Token-> BAMAuthToken: dQfuRMTUxNjc3MjcyNDg1ODppcGFybXM= <- for User : username"
+	c.token = regexp.MustCompile("BAMAuthToken: [^ ]+").FindString(authResp)
+
+	return nil
+}
+
+// Logout Logs out of the current API session.
+// https://docs.bluecatnetworks.com/r/Address-Manager-API-Guide/GET/v1/logout/9.1.0
+func (c *Client) Logout() error {
+	if c.token == "" {
+		// nothing to do
+		return nil
+	}
+
+	resp, err := c.sendRequest(http.MethodGet, "logout", nil, nil)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK {
+		data, _ := io.ReadAll(resp.Body)
+		return fmt.Errorf("status code: %d, message: %s", resp.StatusCode, string(data))
+	}
+
+	authBytes, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return err
+	}
+
+	authResp := string(authBytes)
+	if !strings.Contains(authResp, "successfully") {
+		msg := strings.Trim(authResp, "\"")
```

Comment:
I have somehow developed an aversion against backslashes :)

```suggestion
		msg := strings.Trim(authResp, `"`)
```